driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java

```diff
@@ -177,8 +182,14 @@ protected EncodingMetadata encodeMessageBodyWithMetadata(final BsonOutput bsonOu
             commandStartPosition = bsonOutput.getPosition();
 
             if (payload == null) {
-                addDocument(getCommandToEncode(), bsonOutput, commandFieldNameValidator, null);
+                List<BsonElement> elements = null;
+                if (serverApi != null) {
+                    elements = new ArrayList<>(3);
+                    addServerApiElements(elements);
+                }
+                addDocument(getCommandToEncode(), bsonOutput, commandFieldNameValidator, elements);
             } else {
+                // TODO: should never get to this point with a serverApi specified?
```

Comment:
I think I can remove this TODO.  Any server that supports serverApi will also support OP_MSG so should never get here.